scipy/stats/tests/test_rank.py

```diff
@@ -192,6 +192,74 @@ def test_size_0_axis(self, axis, method, dtype):
         assert_equal(r.shape, shape)
         assert_equal(r.dtype, dtype)
 
+    @pytest.mark.parametrize('axis', range(3))
+    @pytest.mark.parametrize('method', methods)
+    def test_nan_policy_omit_3d(self, axis, method):
+        shape = (20, 21, 22)
+        rng = np.random.default_rng(abs(hash('falafel')))
```

Comment:
Brilliant 😂

Reply:
Hah thanks : ) 

Quick question on this one - I don't need you to review the code, but I just wanted another opinion on the behavior.

Before this PR, NaNs were assigned real ranks greater than all other ranks. 

Now:
![image](https://user-images.githubusercontent.com/6570539/173373823-1a3fc29d-9f84-4f04-aaa6-0a63682f4c22.png)

Can we consider the original behavior a bug since it wasn't documented and isn't really meaningful, so we can change the default behavior a bit here? Currenlly 'omit' is the default because it minimizes impact, but could we change it so that 'propagate' is the default? Do you agree with what 'propagate' does?

Reply:
Wouldn't it be surprising to return nans instead of something for a user? TBH it's difficult for me to judge as I am almost never working with nans. But I think I would expect the function to deal with it and returns something instead of bailing out with nans.

Reply:
_Note: the image above was wrong before. It showed the documentation for `method`, but it was supposed to show the documentation for the new `nan_policy`._

That's OK. I thought you might have an opinion since we've worked together on other NaN PRs.
If it seems surprising to the user, think about how surprising NaN inputs are to the poor function! How is it to divine whether the user considers `np.nan` to be less than or greater than any given number? : )
So `rankdata` will deal with `x = [0, np.nan]` analogously to `np.min` when `nan_policy='propagate'` and analogously to `np.nanmin` when `nan_policy='omit'`.

I'll go ahead and merge since there are three in agreement and one undecided, but I'll open up a related PR so there will still be a chance for others to review. Thanks @AtsushiSakai @tirthasheshpatel @tupui!